### PR TITLE
KAFKA-16524 : Adding a metric for kip 853

### PR DESCRIPTION
**Pull request by muralibasani**
Resolves https://issues.apache.org/jira/browse/KAFKA-16524

- Adding metric number-of-offline-voters

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java [hunk 1]
```diff
@@ -533,6 +533,11 @@ public void initialize(
         // so there are no unknown voter connections. Report this metric as 0.
         kafkaRaftMetrics.updateNumUnknownVoterConnections(0);
 
+        // increment number of offline voters
```
Comment (muralibasani) [inline raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java]: @jsancio need some info on calculating the offline voters

Comment (jsancio) [reply raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java]: When the raft client starts the metrics should be zero. Any replica that is not the leader should report zero. Only the leader should report a non-zero value if there are offline voters.

The best place to implement this is in `LeaderState`. The method `updateReplicaState` is called every time a replica successfully fetches from the leader. You can update that method to also update the number of offline voters.

Comment (muralibasani) [reply raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java]: Updated but in KafkaRaftClient, ptal.

File: raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java [hunk 2]
```diff
@@ -215,6 +215,8 @@ public final class KafkaRaftClient<T> implements RaftClient<T> {
     private volatile RemoveVoterHandler removeVoterHandler;
     private volatile UpdateVoterHandler updateVoterHandler;
 
+    private volatile int numOfOfflineVoters;
```
Comment (jsancio) [inline raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java:218]: This change never writes to this variable.

**Comment by jsancio** [comment]
@muralibasani and I discussed this issue offline. I will be submitting a PR for implementing the metrics.

**Pull request closed without merging**